codex-rs/core/src/codex.rs

```diff
@@ -782,6 +782,36 @@ async fn submission_loop(
                     }
                 });
             }
+
+            Op::SummarizeContext => {
+                let sess = match sess.as_ref() {
+                    Some(sess) => sess,
+                    None => {
+                        send_no_session_event(sub.id).await;
+                        continue;
+                    }
+                };
+
+                // Create a summarization request as user input
+                const SUMMARIZATION_PROMPT: &str = concat!(
+                    "Please provide a summary of our conversation so far, highlighting key points, ",
+                    "decisions made, and any important context that would be useful for future reference. ",
+                    "This summary will be used to replace our conversation history with a more concise ",
+                    "version so choose what details you will need to continue your work. ",
+                    "Provide the summary directly without main title."
+                );
+
+                let summarization_prompt = vec![InputItem::Text {
+                    text: SUMMARIZATION_PROMPT.to_string(),
+                }];
+
+                // Attempt to inject input into current task
+                if let Err(items) = sess.inject_input(summarization_prompt) {
+                    // No current task, spawn a new one
+                    let task = AgentTask::spawn(Arc::clone(sess), sub.id, items);
```

Comment:
Does this work? I think this is the more canonical way to do it.

```suggestion
                    let task = AgentTask::spawn(sess.clone(), sub.id, items);
```

Reply:
This is how it's done here:
https://github.com/openai/codex/blob/bfeb8c92a591e8f20ecabb2a1b5a22e1574e7951/codex-rs/core/src/codex.rs#L711

Reply:
Does what I suggested not work?